### PR TITLE
[stable/concourse] [wip] web hardAntiAffinity

### LINKED ISSUE #15681: [stable/concourse] provide `web.hardAntiAffinity` in `values`
**Is your feature request related to a problem? Please describe.**

We currently allow the configuration of `worker.hardAntiAffinity` to ease the
burden on the user of providing an entire `affinity` rule.

Rather than

```yaml
# manually specifying the whole affinity rule
# (note how you need to know the release name before-hand)
#
worker:
  affinity:
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
      - labelSelector:
          matchLabels:
            app: $RELEASE_NAME-worker
            release: "$RELEASE_NAME"
        topologyKey: kubernetes.io/hostname
```

One can simply

```yaml
worker:
  hardAntiAffinity: true
```

Which is quite convenient.

**Describe the solution you'd like**

Have a `web.hardAntiAffinity` field just like `worker` does.

**Describe alternatives you've considered**

Have a `web.hardAntiAffinity` field just like `worker` does.

**Additional context**

We've seen in [`hush-house`](https://github.com/concourse/hush-house) (our
largest deployment of Concourse leveraging this Helm chart) that it's very
important to have the `web` nodes getting placed in different VMs as with
improvements like [parallel input streaming](https://github.com/concourse/concourse/issues/3992#issuecomment-504036411)
and [the use of zstd for compression](https://github.com/concourse/baggageclaim/pull/22)
we tend now to reach the full available bandwidth of the machines where they
land on.

Thus, having the ability to tell folks to just use `web.hardAntiAffinity: true`
would be great 😁 

Thanks!

### PR DESCRIPTION
**Pull request by aegershman**
#### What this PR does / why we need it:

Ability to configure whether the web pods should be forced (or preferred) to be on different nodes

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #15681

#### Special notes for your reviewer:

Open question: if `web.hardAntiAffinity: false`, this still _prefers_ to schedule the web pods on separate nodes, and there's no way to turn that off (this is also how `workers.hardAntiAffinity` works, too). Are we okay with that...? E.g., is there a reason someone wouldn't want to have the web nodes try to be on different hosts? 

And if someone specifies `podAntiAffinity` under `additionalAffinities`, it'll blow up due to a duplicate key. I believe this is already the case with `worker.hardAntiAffinity`, but nonetheless it's worth pointing out:

```yml
web:
  hardAntiAffinity: false
  additionalAffinities:
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
      - labelSelector:
          matchLabels:
            app: concourse-web-web
            release: "concourse-web"
        topologyKey: kubernetes.io/hostname
```

would result in...

```yml
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchLabels:
                app: concourse-web-web
                release: concourse-web
            topologyKey: kubernetes.io/hostname
        
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 100
            podAffinityTerm:
              topologyKey: kubernetes.io/hostname
              labelSelector:
                matchLabels:
                  app: concourse-web-web
                  release: "concourse-web"
```

... which is maybe okay?

I'm not sure what the best course of action is, because it's hard for me to imagine needing an affinity other than "try not to clump up onto a single host"? This same logic applies to the concourse workers. But I'll leave this decision up to someone else.

This might also be slightly problematic since the `web` pods use a `deployment`, which won't take too kindly to having `hardAntiAffinity` on while trying to do a rolling upgrade? Haven't thought about this too much though, it might be fine.

Also please let me know if you'd like any doc changed, formatting, etc.

And finally, please keep in mind I haven't tested this in a real production environment yet, just on a local machine, so I need to get around to doing that at some point.

TL;DR, don't merge this yet, haven't tested this in prod or deeply thought through implications

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by aegershman** [comment]
Going to go ahead and close this for the time being since I don't have a whole lot of time to dedicate to thinking on it, anyone else interested in taking this up please do 👍 thanks!
